config/routes.rb

```diff
@@ -10,4 +12,12 @@
 
   get '/genre/:id', to: 'categories#show'
 
+  # registration/authentication
+  get '/register', to: 'users#new'
+  get '/login', to: 'sessions#new'
+  post '/login', to: 'sessions#create'
+  get '/logout', to: 'sessions#destroy'
+
+  resources :users, only: [:show, :create, :edit, :update]
```

Comment:
Where are those methods `:show` and `:edit`, `:update` ?  So far you should not open new routes to stuff you have not built or secured.